saleor/graphql/product/mutations/products.py

```diff
@@ -1300,14 +1295,20 @@ def clean_input(
             # We need to transform them into the format they're stored in the
             # `Product` model, which is HStore field that maps attribute's PK to
             # the value's PK.
-            attributes = cleaned_input.get("attributes", [])
+            attributes = cleaned_input.get("attributes")
             try:
-                cls.validate_duplicated_attribute_values(
-                    attributes, used_attribute_values, instance
-                )
-                cleaned_input["attributes"] = cls.clean_attributes(
-                    attributes, product_type
-                )
+                if attributes:
+                    cls.validate_duplicated_attribute_values(
+                        attributes, used_attribute_values, instance
+                    )
+                    cleaned_input["attributes"] = cls.clean_attributes(
+                        attributes, product_type
+                    )
+                elif not instance.pk and not attributes:
```

Comment:
Please add a comment here. It's hard to understand what happened at first sight.